src/modules/azure.rs

```diff
@@ -53,28 +74,44 @@ pub fn module<'a>(context: &'a Context) -> Option<Module<'a>> {
     Some(module)
 }
 
-fn get_azure_subscription_name(context: &Context) -> Option<SubscriptionName> {
+fn get_azure_profile_info(context: &Context) -> Option<Subscription> {
     let mut config_path = get_config_file_location(context)?;
     config_path.push("azureProfile.json");
 
-    let parsed_json = parse_json(&config_path)?;
-
-    let subscriptions = parsed_json.get("subscriptions")?.as_array()?;
-    let subscription_name = subscriptions.iter().find_map(|s| {
-        if s.get("isDefault")? == true {
-            Some(s.get("name")?.as_str()?.to_string())
+    if config_path.exists() {
+        let azure_profile: Option<AzureProfile> = load_azure_profile(&config_path);
+
+        if let Some(azure_profile) = azure_profile {
+            let subscription = azure_profile.subscriptions.iter().find_map(|s| {
+                if s.is_default {
+                    Some(s.clone())
+                } else {
+                    None
+                }
+            });
+            subscription
         } else {
             None
         }
-    });
-    if subscription_name.is_some() {
-        subscription_name
     } else {
-        log::info!("Could not find subscription name");
         None
     }
 }
 
+fn load_azure_profile(config_path: &PathBuf) -> Option<AzureProfile> {
+    let json_data = fs::read_to_string(&config_path).expect("Unable to open azureProfile.json");
+    let sanitized_json_data = json_data.strip_prefix('\u{feff}').unwrap_or(&json_data);
+    let azure_profile: Option<AzureProfile> =
+        if let Ok(azure_profile) = serde_json::from_str::<AzureProfile>(sanitized_json_data) {
+            Some(azure_profile)
+        } else {
+            log::info!("Failed to parse azure profile.");
+            None
+        };
+
+    azure_profile
+}
```

Comment:
No need to use `expect` here

```suggestion
fn load_azure_profile(config_path: &PathBuf) -> Option<AzureProfile> {
    let json_data = fs::read_to_string(&config_path).ok()?;
    let sanitized_json_data = json_data.strip_prefix('\u{feff}').unwrap_or(&json_data);
    if let Ok(azure_profile) = serde_json::from_str::<AzureProfile>(sanitized_json_data) {
        Some(azure_profile)
    } else {
        log::info!("Failed to parse azure profile.");
        None
    }
}
```